### PR TITLE
Tundra beat: VSCode auth design changes

### PR DESCRIPTION
**Pull request by oliviacpu**
## Links
* Remix link: https://tundra-beat.glitch.me/vscode-auth

## Changes:
* Redesign of the VSCode auth page
* Supports OAuth and Magic Code options currently

## How To Test:
* Try the OAuth and Magic Code options and make sure it opens VSCode

### CODE REVIEW

File: src/components/layout/sign-in-layout.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import useDevToggle from 'State/dev-toggles';
+
+import styles from './sign-in-layout.styl';
+
+const keyImageUrl = 'https://cdn.glitch.com/8ae9b195-ef39-406b-aee0-764888d15665%2Foauth-key.svg?1544466885907';
+
+const TermsAndConditions = () => (
+  <div className={styles.termsAndConditions}>
+    By signing into Glitch, you agree to our <Link to="/legal/#tos">Terms of Services</Link> and <Link to="/legal/#privacy">Privacy Statement</Link>
```
Comment (clottman) [inline src/components/layout/sign-in-layout.js]: Should probably be *Terms of Service* (not Services)

Comment (clottman) [reply src/components/layout/sign-in-layout.js]: We call `legal/#privacy` our Privacy Policy, not Statement

File: src/components/sign-in/password-login.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import Notification from 'Components/notification';
+
+import useDebouncedValue from 'Hooks/use-debounced-value';
+
+import { useAPI } from 'State/api';
+import { useCurrentUser } from 'State/current-user';
+
+import styles from './styles.styl';
+
+function useEmail() {
```
Comment (clottman) [inline src/components/sign-in/password-login.js]: Looks like the same function is being used in get-magic-code.js - can we move it to a helper/shared file?

File: src/components/sign-in/use-magic-code.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        <Notification persistent type="success">
+          Success!
+        </Notification>
+      )}
+      {status === 'error' && (
+        <>
+          <Notification persistent type="error">
+            Error
+          </Notification>
+          <div>Code not found or already used. Try signing in with email.</div>
```
Comment (clottman) [inline src/components/sign-in/use-magic-code.js]: Does this need to be `<div>`, or can it be `<p>`?

Comment (taravancil) [reply src/components/sign-in/use-magic-code.js]: If `<p>`, we should use our `<Text>` component

**Comment by oliviacpu** [comment]
The copy changes and extracting the function will be part of work after this to unify all the sign-in efforts. I think I fixed the redirect and used the `<Text>` component thing.

**Comment by oliviacpu** [comment]
`This redirects to Glitch.com when I test it in my browser`

Setting `destinationAfterAuth` on Glitch.com local storage to the value from the remix local storage before you do the OAuth flow should "fix" this until it is deployed I think

**Pull request merged**